js/store.js

```diff
@@ -29,31 +44,39 @@ const store = {
     return null;
   },
 
-  setFirstParty(param) {
-    /**
-     * @param object with hostname as key
-     * @todo validate the param.-
-     * @todo code to be updated in next PR
-     */
-    return this.set(param);
+  setFirstParty(hostname, data) {
+    if (!hostname) {
+      throw new Error('setFirstParty requires a valid hostname argument');
+    }
+
+    const websites = clone(this._websites);
+    websites[hostname] = data;
+    this._write(websites);
   },
 
-  async set(websites) {
-    /**
-      * @todo code to be updated in next PR
-    */
-    return await browser.storage.local.set({ websites });
+  setThirdParty(origin, target, data) {
```

Comment:
@princiya , @jonathanKingston : Thank you for the feedback! Capture.js is actually passing in `document` currently as the first argument for this method, not `origin`. I used `document` because I identify third party requests by comparing `document` to `target`.

What is meant by `origin` in this context? Is it the same as `origin` as defined in Issue #42 ?

Using `origin` here seems to be in contrast with why we're interested in `document` and `target` as explained in Issue #42 . What am I missing?

I agree that we could use JSDoc to help clarify these API questions as well as part of the scripts (mine included)!